specs/Standards/NonFungibleToken/Payout.md

```diff
@@ -66,35 +72,43 @@ Financial contracts MAY take a cut of the NFT sale price as commission, subtract
 #[derive(Serialize, Deserialize)]
 #[serde(crate = "near_sdk::serde")]
 pub struct Payout {
-  pub payout: HashMap<AccountId, U128>,
+  payout: HashMap<AccountId, U128>,
 }
 
-pub trait Payouts{
+pub trait Payouts {
   /// Given a `token_id` and NEAR-denominated balance, return the `Payout`.
   /// struct for the given token. Panic if the length of the payout exceeds
   /// `max_len_payout.`
-  fn nft_payout(&self, token_id: String, balance: U128, max_len_payout: u32) -> Payout;
+  fn nft_payout(&self, token_id: String, balance: U128, max_len_payout: Option<u32>) -> Payout;
   /// Given a `token_id` and NEAR-denominated balance, transfer the token
   /// and return the `Payout` struct for the given token. Panic if the
   /// length of the payout exceeds `max_len_payout.`
-  #[payable]
```

Comment:
> Also the `payable` decorator is removed since the market place handles the transfers.

I don't understand this statement. `payable` attribute guards against using function-call-only access keys, and given that `nft_transfer_payout` is a sibling to `nft_transfer` which is guarded, I believe it is a mistake to drop `payable` here.

Reply:
Ah good point! I think at the time I thought it was because the caller should handle the payout and that the caller would always be another contract, but this isn't true.